libmamba/src/specs/version.cpp

```diff
@@ -217,78 +216,108 @@ namespace mamba::specs
             // We do a lexicographic compare with an infite sequence of empties
-            if ((first1 != last1))
+            if ((iter1 != last1))
             {
-                for (; first1 != last1; ++first1)
+                for (; iter1 != last1; ++iter1)
                 {
-                    if (auto c = comp(*first1, empty); c != strong_ordering::equal)
+                    if (auto c = comp(*iter1, empty2); c != strong_ordering::equal)
                     {
-                        return c;
+                        return { c, static_cast<std::size_t>(std::distance(first1, iter1)) };
                     }
                 }
             }
             // first2 != last2
             // They have the same leading elements but 2 has more elements
             // We do a lexicographic compare with an infite sequence of empties
-            if ((first2 != last2))
+            if ((iter2 != last2))
             {
-                for (; first2 != last2; ++first2)
+                for (; iter2 != last2; ++iter2)
                 {
-                    if (auto c = comp(empty, *first2); c != strong_ordering::equal)
+                    if (auto c = comp(empty1, *iter2); c != strong_ordering::equal)
                     {
-                        return c;
+                        return { c, static_cast<std::size_t>(std::distance(first2, iter2)) };
                     }
                 }
             }
             // They have the same elements
-            return strong_ordering::equal;
+            return { strong_ordering::equal, static_cast<std::size_t>(std::distance(first1, iter1)) };
+        }
+
+        template <typename Iter1, typename Iter2, typename Empty, typename Cmp>
+        constexpr auto lexicographical_compare_three_way_trailing(
+            Iter1 first1,
+            Iter1 last1,
+            Iter2 first2,
+            Iter2 last2,
+            Empty empty,
+            Cmp comp
+        ) -> std::pair<strong_ordering, std::size_t>
+        {
+            return lexicographical_compare_three_way_trailing(
+                first1,
+                last1,
+                first2,
+                last2,
+                empty,
+                empty,
+                comp
+            );
         }
 
         template <>
         auto compare_three_way(const VersionPart& a, const VersionPart& b) -> strong_ordering
         {
             return lexicographical_compare_three_way_trailing(
-                a.cbegin(),
-                a.cend(),
-                b.cbegin(),
-                b.cend(),
-                VersionPartAtom{},
-                [](const auto& x, const auto& y) { return compare_three_way(x, y); }
-            );
+                       a.cbegin(),
+                       a.cend(),
+                       b.cbegin(),
+                       b.cend(),
+                       VersionPartAtom{},
+                       [](const auto& x, const auto& y) { return compare_three_way(x, y); }
```

Comment:
why this extra indentation?

Reply:
Not sure why clang-format decided to change that. The function is now returning a pair and only the first member is used (note the `.first` on the next line).
We could `auto [cmp, idx] = lexicographical_compare_three_way_trailing(...); return cmp;` if you think it is better.

Reply:
I think the current implementation is fine, I was just surprised by this double indentation of arguments in a function call. We can try to fix the clang-format configuration later, no need to change the code for that imo.

Reply:
Yes, clang-format does very obscure things sometimes. I could not find any configuration for this.